src/Text/Parsing/StringParser/CodeUnits.purs

```diff
@@ -62,10 +64,24 @@ anyDigit = try do
 
 -- | Match the specified string.
 string :: String -> Parser String
-string nt = Parser \s ->
```

Comment:
I think if the user supplies a string containing a lone surrogate to `string`, so that returning that string as the parse result splits a surrogate pair in half, I think we should actually split the surrogate pair in that case (since that's what the user has asked us to do). Come to think of it, we should have tests for that.

Also, this implementation looks like it behaves the same as the old one? The result of `findIndex` isn't used, and both branches of the `case` statement look the same to me.

Reply:
> Also, this implementation looks like it behaves the same as the old one? The result of findIndex isn't used, and both branches of the case statement look the same to me.

I simplified it in the latest commit. Not sure whether that is a naive implementation, but tests still pass.

Reply:
Sorry, I mean I would rather revert the implementation of `string` back to how it is in this repo's `main` branch right now.

Reply:
Ah, gotcha. I implemented mine because I thought the code points would mess things up. However, I realize now that it doesn't.